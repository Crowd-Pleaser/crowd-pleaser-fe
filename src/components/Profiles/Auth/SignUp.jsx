import React from 'react';
import { useSignUp } from '../../../hooks/authHooks';
import styles from './Styling.css';

export default function SignUp() {
  const {
    loading,
    firstName,
    lastName,
    email,
    role,
    password,
    handleChange,
    handleSignUp
  } = useSignUp();

  return (
    <div className={styles.Form}>
      <h2>Sign Up</h2>
      <form onSubmit={handleSignUp}>
        <input type="text" value={firstName} name="firstName" onChange={handleChange} placeholder="First name" />
        <input type="text" value={lastName} name="lastName" onChange={handleChange} placeholder="Last name" />
        <input type="text" value={email} name="email" onChange={handleChange} placeholder="Email address" />
        <input type="password" value={password} name="password" onChange={handleChange} placeholder="Create password" />
        <label htmlFor="user">User</label>
        <input type="radio" value="User" name="role" id="user" onChange={handleChange} checked={role === 'User'}/>
        <label htmlFor="restaurant">Restaurant Owner</label>
        <input type="radio" value="Restaurant" name="role" id="restaurant" onChange={handleChange} checked={role === 'Restaurant'} />
        <button disabled={loading}>Sign Up</button>
      </form>
    </div>
  );
}
