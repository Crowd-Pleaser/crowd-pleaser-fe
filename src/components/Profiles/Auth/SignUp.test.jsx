import React from 'react';
import { shallow } from 'enzyme';
import SignUp from './SignUp';

describe('SignUp', () => {
  it('matches a SignUp snapshot', () => {
    const wrapper = shallow(<SignUp />);
    expect(wrapper).toMatchSnapshot();
  })
})