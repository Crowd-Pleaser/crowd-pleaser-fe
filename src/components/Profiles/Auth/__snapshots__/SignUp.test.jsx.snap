// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp matches a SignUp snapshot 1`] = `
<div>
  <h2>
    Sign Up
  </h2>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="First name"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Last name"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Email address"
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Create password"
      type="password"
      value=""
    />
    <label
      htmlFor="user"
    >
      User
    </label>
    <input
      checked={true}
      id="user"
      name="role"
      onChange={[Function]}
      type="radio"
      value="User"
    />
    <label
      htmlFor="restaurant"
    >
      Restaurant Owner
    </label>
    <input
      checked={false}
      id="restaurant"
      name="role"
      onChange={[Function]}
      type="radio"
      value="Restaurant"
    />
    <button>
      Sign Up
    </button>
  </form>
</div>
`;
